forte/proc/aset2.py

```diff
@@ -0,0 +1,309 @@
+# @BEGIN LICENSE
+#
+# forte_inversion by Psi4 Developer, a plugin to:
+#
+# Psi4: an open-source quantum chemistry software package
+#
+# Copyright (c) 2007-2016 The Psi4 Developers.
+#
+# The copyrights for code used from other parties are included in
+# the corresponding files.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# @END LICENSE
+#
+
+import psi4
+import forte
+import numpy as np
+from forte.proc.dsrg import ProcedureDSRG
+
+def aset2_driver(state_weights_map, scf_info, ref_wfn, mo_space_info, options):
```

Comment:
I wonder if this could be a Python class and make the rest of the file more elaborate.